lib/fibonacci.rb

```diff
@@ -1,8 +1,19 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 # Hint, you may want a recursive helper method
 def fibonacci(n)
```

Comment:
This works, but you're maintaining the list of all the Fibonacci numbers from 1 to n the entire length of the recursion.  You only need the last two recursive numbers.